### ISSUE TITLE
SkipToContent property href does not work with HashRouter

### ISSUE THREAD

**Issue by nxn-4-wdf** [carbon-design-system/carbon#3698]
## Environment

> Operating system

Any.

> Browser

Any.

## Detailed description

> What version of the Carbon Design System are you using?
10.4

As instructed in the Carbon tutorial, app uses a UI-Shell with header navigation and `react-router-dom` Link components, which lead to different pages sorted by a HashRouter. In order to enable navigation to internal page anchors, page also uses HashLink components from `react-router-hash-link`.

> What did you expect to happen?

With keyboard navigation, pressing on the Enter-key when on the SkipToContent should remain on the same page and scroll to the HTML element with `id="main-content"`.

> What happened instead?

Application tries to navigate to `/#/main-content` which renders the default Router page, instead of staying on the same page.

## Steps to reproduce the issue

Example application on CodeSandBox: https://37eph.csb.app/

From there, using keyboard navigation ("Tab" and "Shift-Tab"):
1. Select header navigation `Page two` and press the Enter-key
2. With Shift-Tab combination, get back to display SkipToContent element. Partial screenshot:
![image](https://user-images.githubusercontent.com/52698706/62766501-eb77cf80-ba92-11e9-8a58-fab2490e1e2c.png)
3. Press the Enter-key: page is changed (NOT expected), and as a side-effect, scroll amount is wrong (page heading is masked by the header bar)
![image](https://user-images.githubusercontent.com/52698706/62766622-2bd74d80-ba93-11e9-9142-22f86d821c7e.png)

> Please create a reduced test case in CodeSandbox

Code is here : https://codesandbox.io/s/skiptocontent-37eph

## Additional information

- Screenshots or code: see above

**Comment by dakahn** [comment]
Hello friend! I think this issue should be filed against the Carbon Tutorial repository.

**Comment by abbeyhrt** [comment]
I moved this issue to the Carbon tutorial. You can follow it's progress in the referenced [issue](https://github.com/carbon-design-system/carbon-tutorial/issues/2810)

**Comment by st004547** [comment]
For anyone else looking up this issue:

Using `react-router-dom`, we weren't able to use the `SkipToContent` component since it does not use the `Link` component from `react-router-dom`. Instead, we just used the `bx--skip-to-content` style on `Link`.

There's also a concern that the link does not actually bring keyboard focus to the target area, which defeats the usefulness for keyboard navigation. The issue is discussed here: https://www.bignerdranch.com/blog/web-accessibility-skip-navigation-links/
